resolwe/flow/managers/base.py

```diff
@@ -353,15 +356,29 @@ def _prepare_context(self, data_id, data_dir, runtime_dir, **kwargs):
         }
 
         # extend the settings with whatever the executor wants
-        self.executor.extend_settings(data_id, files)
+        self.executor.extend_settings(data_id, files, extra)
```

Comment:
On the other hand, the docstring for `extended_settings` specifies this plainly as a field for general raw files (as opposed to JSON serializations in the `files` parameter). Secrets are just one kind of opaque blob, so usage agrees with the docstring's specification.

Reply:
But we also prevent listing of such dirs and restrict masks of files. So it may have unwanted consequences if you put such file directly into executor dir.
I would prefer to narrow use of this, make it more securely (it already is) and clearly state that.